internal/keycloakb/archivedbmodule.go

```diff
@@ -0,0 +1,49 @@
+package keycloakb
+
+import (
+	"context"
+
+	"github.com/cloudtrust/common-service/database/sqltypes"
+	"github.com/cloudtrust/common-service/log"
+	"github.com/cloudtrust/common-service/security"
+)
+
+const (
+	storeUserArchiveStmt = `
+	  INSERT users (realm_name, user_id, timestamp, details)
+	  VALUES (?, ?, UTC_TIMESTAMP, ?)
+	`
+)
+
+// ArchiveDBModule interface
+type ArchiveDBModule interface {
+	StoreUserDetails(ctx context.Context, realm, userID string, details []byte) error
+}
+
+type archiveDBModule struct {
+	db     sqltypes.CloudtrustDB
+	cipher security.EncrypterDecrypter
+	logger log.Logger
+}
+
+// NewArchiveDBModule returns an archive DB module.
+func NewArchiveDBModule(db sqltypes.CloudtrustDB, cipher security.EncrypterDecrypter, logger log.Logger) ArchiveDBModule {
+	return &archiveDBModule{
+		db:     db,
+		cipher: cipher,
+		logger: logger,
+	}
+}
+
+func (a *archiveDBModule) StoreUserDetails(ctx context.Context, realm, userID string, details []byte) error {
```

Comment:
I think it would make sense we provide a UserRepresentation to this module and it will be module responsibility to convert it into byte array.

Reply:
see next answer